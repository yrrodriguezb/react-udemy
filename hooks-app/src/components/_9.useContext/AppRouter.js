import React from 'react';
import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom";
import { AboutScreen } from './AboutScreen'
import { LoginScreen } from './LoginScreen'
import { HomeScreen } from './HomeScreen'
import { Navbar } from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <React.Fragment>

        <Navbar />

        <Routes>
          <Route path='/' element={ <HomeScreen /> } />
          <Route path='/about' element={ <AboutScreen /> } />
          <Route path='/login' element={ <LoginScreen /> } />
          <Route path='*' element={ <HomeScreen /> } />
        </Routes>

      </React.Fragment>
    </Router>
  )
}
