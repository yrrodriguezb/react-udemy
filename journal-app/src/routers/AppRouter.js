import React, { useEffect, useState } from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import { useDispatch } from 'react-redux';
import { firebase } from "../firebase/firebase.config";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { login } from '../actions/auth';
import { Spinner } from '../components/common/spinner';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

  const dispatch = useDispatch();

  const [ checking, setChecking ] = useState(true);
  const [ isLoogedIn, setIsLoogedIn ] = useState(false)

  useEffect(() => {
   firebase.auth().onAuthStateChanged( user => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setIsLoogedIn(true);
      } else {
        setIsLoogedIn(false);
      }
      
      setChecking(false);
   });
  }, [ dispatch, setChecking, setIsLoogedIn ])
  

  if (checking) {
    return <Spinner />
  }

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute 
            path="/auth"
            isAuthenticated={ isLoogedIn }
            component={ AuthRouter }
          />
          <PrivateRoute 
            exact
            isAuthenticated={ isLoogedIn }
            path="/"
            component={ JournalScreen }

          />
          <Redirect 
            to="/auth/login" 
          />
        </Switch>
      </div>
    </Router>
  )
}
