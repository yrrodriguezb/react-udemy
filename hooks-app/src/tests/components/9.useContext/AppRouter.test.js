import React from 'react';
import '@testing-library/jest-dom'
import { mount } from "enzyme";
import { AppRouter } from '../../../components/_9.useContext/AppRouter';
import { UserContext } from '../../../components/_9.useContext/UserContext';

describe('Pruebas en <AppRouter />', () => {
  const user = {
    name: 'Yeison Roriguez',
    email: 'yrrodriguez@gmail.com'
  }

  const wrapper = mount(
    <UserContext.Provider value={ { user } }>
      <AppRouter />
    </UserContext.Provider>
  );

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
})