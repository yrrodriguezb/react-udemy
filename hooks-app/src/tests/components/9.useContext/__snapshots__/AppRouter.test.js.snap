// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-primary"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="p-5"
  >
    <h1>
      HomeScreen
    </h1>
    <hr />
    <pre
      className="container"
    >
      {
    "name": "Yeison Roriguez",
    "email": "yrrodriguez@gmail.com"
}
    </pre>
  </div>,
]
`;
